### ISSUE TITLE
Resizing desktop browsers introduces small unnecessary horizontal scrolling on a few pages

### ISSUE THREAD

**Issue by johnlettman** [canonical/canonical.com#976]
## Summary
Portions of the Canonical company website include an errant right margin when desktop browsers, such as Brave Browser or Chromium, are sized to a width below 1300px. As a result, a minor unnecessary horizontal scrollbar is introduced to the page, stealing away a few vertical pixels.

It appears to be leftover margin or padding since the horizontal scrolling only reveals a few pixels to the right, filled with empty space.

Impacted:
- https://canonical.com
- https://canonical.com/partners/desktop
- https://canonical.com/partners/channel-and-reseller
- https://canonical.com/partners/iot-device
- https://canonical.com/partners/ihv-and-oem

NOT impacted:
- https://canonical.com/projects
- https://canonical.com/partners/become-a-partner
- https://canonical.com/partners/find-a-partner
- https://canonical.com/partners/executive-summit 
- and so on...
- https://canonical.com/careers
- and so on...

## Process

1. Open https://canonical.com in a desktop browser, e.g. Brave Browser or Chromium
2. Resize the browser horizontally to below 1300 pixels
3. Observe the introduction of a horizontal scrollbar

## Current and expected result

As with the non-impacted pages, the horizontal scrollbar should not appear.

### Brave Browser
![Screenshot_20230719_124641](https://github.com/canonical/canonical.com/assets/20583294/6377873e-2f0e-4470-a258-fc6fb3ce1f61)

### Chromium (clean install)
![Screenshot_20230719_130010](https://github.com/canonical/canonical.com/assets/20583294/2dc5bc4a-e5f9-4e6b-9915-209827df9f7b)

### Brave Browser
Brave 1.52.129 Chromium: 114.0.5735.198 (Official Build) (64-bit) 
- **Revision:** `c3029382d11c5f499e4fc317353a43d411a5ce1c-refs/branch-heads/5735@{#1394}`
- **JavaScript Engine:**  `V8 11.4.183.25`
- **User Agent:** `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36`

### Chromium (clean install)
Chromium 114.0.5735.198 (Official Build) snap (64-bit)
- **Revision:** `c3029382d11c5f499e4fc317353a43d411a5ce1c-refs/branch-heads/5735@{#1394}`
- **JavaScript Engine:** `V8 11.4.183.25`
- **User Agent:** `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36`

**Comment by bartaz** [comment]
Seems to be related to custom styling of logo section component:

https://github.com/canonical/canonical.com/blob/27955fe1f6ebf15415b1c0902ed35614285de81a/static/sass/styles.scss#L722-L740

**Comment by johnlettman** [comment]
I spent some time in the inspector but couldn't locate the element causing the horizontal scroll.
Spanning the entire vertical height of the induced whitespace, no components trigger the element picker other than the `<html>` tag. None of the CSS styling applied there seems to trigger it. 

Checking per:

> Seems to be related to custom styling of logo section component:
> 
> https://github.com/canonical/canonical.com/blob/27955fe1f6ebf15415b1c0902ed35614285de81a/static/sass/styles.scss#L722-L740

I was not able to locate `.p-logo-section` classed elements, but it's entirely possible they're renamed in the resulting build.

It's an unusual one for me. :sweat_smile:

**Comment by johnlettman** [comment]
Okay, it is that @bartaz nice find!

Here, there is a CSS rule that introduces the issue:
```css
.p-logo-section__items {
    margin-left: -2rem;
    width: calc(100% + 2rem);
}
```

It hails from a pseudo-URL reference (sourcemap?):
`https://canonical.com/node_modules/vanilla-framework/scss/_patterns_lists.scss`

Normally, the width `+ 2rem` in combination with `margin-left: -2rem` could cancel each other out; but, the following line is overriding the `-2rem`:

https://github.com/canonical/canonical.com/blob/27955fe1f6ebf15415b1c0902ed35614285de81a/static/sass/styles.scss#L725

Removing the `+ 2rem` from the `width` calculation on the media-queried `.p-logo-section__items` resolves the issue. The alternative of dropping line 725 on the `styles.scss` results in an undesirable appearance.

I recommend adding to `.p-logo-section__items` an override for the `width` calculation to force it as `calc(100%)`.

**Comment by lyubomir-popov** [comment]
@carkod the -2rem is no longer needed, might be worth cleaning that up too.

**Comment by johnlettman** [comment]
@lyubomir-popov
The `margin-left: -2rem` and `width: calc(100% -2rem)` in `.p-logo-section__items` is in the upstream Canonical [vanilla-framework](https://github.com/canonical/vanilla-framework) project included here:
https://github.com/canonical/canonical.com/blob/27955fe1f6ebf15415b1c0902ed35614285de81a/package.json#L38

The section introducing the problematic margin and width CSS rules is [located on lines 111 to 122 in `main/scss/_patterns_logo-section.scss`](https://github.com/canonical/vanilla-framework/blob/main/scss/_patterns_logo-section.scss#L111-L122).

It's overridden to 0 here:
https://github.com/canonical/canonical.com/blob/27955fe1f6ebf15415b1c0902ed35614285de81a/static/sass/styles.scss#L725

It may be a consideration to make the changes here (to drop the override and revert my PR) and on the upstream framework simultaneously.

**Comment by johnlettman** [comment]
@lyubomir-popov Something like this?
https://github.com/johnlettman/vanilla-framework/commit/0e55ac6bbb18fcb2207283df37338d0331f6335e

**Comment by lyubomir-popov** [comment]
@johnlettman not exactly, we made some changes to the logo section, to make it work with logos that have a fixed height, but arbitrary width. This is currently local styles in the canonical.com repo. Part of it was flipping the margin from left to right, but I must have missed the -2rem compensation to bring the first item on a row. So that needs to be set to 0 too. THen we need to upstream all of this to vanilla, but as a separate variant, so that we don't affect existing logo strips - as the logos themselves need to be resized differently. I'd leave this to the vanilla team to upstream, all I meant was that we can remove the negativfe margin locally here, so it is tidy when the vanilla team pick it up for upstreaming.
